File: 프로그래머스/lv2/87390. n＾2 배열 자르기/n＾2 배열 자르기.js
function solution(n, left, right) {
    return makeArr(n, left, right);
}

const makeArr = (n, left, right) => {
    const arr = [];
    
    const 시작행번호 = Math.floor(left/n); 
    const 끝행번호 = Math.floor(right/n); 
    
    for(let i = 시작행번호;  i <= 끝행번호; i++) {
        for(let j = 1; j <= n; j++) {
            if(j <= (i + 1)) arr.push(i + 1);
            else arr.push(j)
        }
    }
    
    return arr.slice(left%n, (끝행번호 - 시작행번호)*n + right % n + 1);
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
import math

def solution(n, left, right):
    start_turn = math.floor(left / n)
    end_turn = math.floor(right / n)
    
    s = []
    for i in range(start_turn, end_turn + 1):
        for j in range(n):
            if j <= i:
                s.append(i + 1)
            else:
                s.append(j + 1)
    
    res = s[left % n:(end_turn - start_turn) * n + (right % n) + 1]
    
    return res